chainreg/chainregistry.go

```diff
@@ -840,3 +864,28 @@ var (
 		},
 	}
 )
+
+// checkOutboundPeers checks the number of outbound peers connected to the
+// provided RPC client. If the number of outbound peers is below 6, a warning
+// is logged. This function is intended to ensure that the chain backend
+// maintains a healthy connection to the network.
+func checkOutboundPeers(client *rpcclient.Client) error {
+	peers, err := client.GetPeerInfo()
+	if err != nil {
+		return err
+	}
+
+	var outboundPeers int
+	for _, peer := range peers {
+		if !peer.Inbound {
+			outboundPeers++
+		}
+	}
+
+	if outboundPeers < 6 {
+		log.Warnf("The number of outbound peers (%d) is "+
```

Comment:
This should give a bit more context on what is actually checked. So perhaps something like "The chain backend has an insufficient number of connected outbound peers (%d connected, expected minimum is 6) which can be a security issue. Connect to more trusted nodes manually if necessary."

Reply:
Thanks that is definitely better! I love this proactive log message instead of the current reactive one!